lib/bolt/cli.rb

```diff
@@ -248,8 +251,8 @@ def parse
 
       @config.load_file(options[:configfile])
       @config.update_from_cli(options)
-      Logger.configure(@config)
       @config.validate
+      Logging.logger[:root].level = @config[:log_level] || :notice
```

Comment:
By doing this we don't have to worry about loggers that may already be created in other class instances?

Reply:
My understanding is: Because the only appender is defined on the root logger, logging to any other logger will do nothing directly. But because the other loggers are additive, they will also pass the event up to the root logger, which will then choose whether to log it based on this level.

But based on that understanding, it doesn't seem like the thing we do to set the net-ssh logging to `warn` should work (since it should still be passed up to the root logger, which will log it because it's at notice level)... so let me investigate further.

Reply:
Okay, that's not quite true.

It turns out that the logger that initially receives the log method (`debug()`, `warn()`) decides whether it will be logged. When passing the event up its parent chain (which it will only do if the event was at an appropriate 
 level), it skips checking the level of its parent loggers and will log unconditionally. All of that is true *unless* a specific appender has been configured with a less verbose log level than its logger. However, the default log level for all appenders is `all`, meaning they will log any message given to them, deferring all decisions about level to the logger itself.

Changing the log level on a logger will automatically propagate the change to its children, assuming they haven't *explicitly* set a level.

The short version is that this line of code will affect all loggers that don't specify a log level, whether they've been created yet or not.

Reply:
cool